src/gtclang/Frontend/ClangASTExprResolver.cpp

```diff
@@ -446,9 +464,20 @@ class StorageResolver {
         }
 
         // Apply the offset
-        if(currentArgumentKind_ == CK_Dimension)
+        if(currentArgumentKind_ == CK_Dimension) {
           offset_[curDim_] += offset;
-        else
+          if((legalDimensions_[curDim_] == 0) && (offset_[curDim_] != 0)) {
+            std::string dimensionOutput;
+            dimensionOutput += legalDimensions_[0] ? "i " : "";
+            dimensionOutput += legalDimensions_[1] ? "j " : "";
+            dimensionOutput += legalDimensions_[2] ? "k " : "";
+            dimensionOutput.pop_back();
```

Comment:
we insert to later pop_back?

Reply:
pop_back just removes the trailing whitespace.
This is just a simple way of doing it where you don't need to look ahead/behind to figure out where you need to write the whitespace as we can have any number of variables and everyone could be either the first or the last